### DISCUSSION TITLE
Nutzung mit Blazor / .NET and C#

### DISCUSSION THREAD

**Discussion by deleonio** [public-ui/kolibri#1527] [Q&A] [answered]
In diesem Thread wollen wir die Verwendung von KoliBri mit Blazor thematisieren. Mehr dazu in Kürze.

**Comment by deleonio** [comment]
@laske185 bitte stelle uns ein Beispiel bereit. Danke

**Comment by MarcusMorba** [comment]
@deleonio & @laske185 - was mich aktuell betrifft, hat sich das Thema ".Net"-Integration erledigt, da ich aus dem Projekt ausgestiegen bin.

Meine Vorab-Recherchen dazu haben folgendes ergeben: .Net v6.0 (LTS) wird wohl noch in so einigen Behörden verwendet und ist eigentlich veraltet. Deshalb fehlen dort auch viele Features der neueren Versionen inkl. der Features für eine verbesserte Accessibility. 

Deshalb sollte v6.0 nicht mehr für zukünftige Anwendungen verwendet werden, besser schon die v8 Beta mit LTS (36 Monate ab Nov 2023). EOS für .Net v6.0 LTS ist der 12. Nov. 2024. Für .Net v7.0 ist es der 14 May 2024.
Quelle: https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

Generell sollte gelten: Es wäre besser zukünftige Anwendung im Bereich Desktop-App, Web-App (PWA) und Web-Service mit moderneren Mitteln neu zu entwickelt , anstatt zu versuchen Altlasten auf einen aktuellen Stand zubringen. Alternativen zu .Net sollten entsprechend der Anforderungen und vor allem für langfristige Projekte in Betracht gezogen werden. 

Einige Resourcen
- https://learn.microsoft.com/en-us/windows/web/ 
- https://learn.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility (10/13/2022)
- https://learn.microsoft.com/en-us/windows/win32/accessibility/accessibility (01/07/2021)
- https://learn.microsoft.com/en-us/dotnet/framework/ui-automation/accessibility-best-practices (09/15/2021)
- https://learn.microsoft.com/en-us/dotnet/desktop/winforms/windows-forms-accessibility-improvements?view=netframeworkdesktop-4.8 (11/06/2020)
- https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium (01/24/2023)

**Reply by deleonio** [reply]
@laske185 kannst Du bitte auf diese Nachricht antworten und gerne notwendige Schritte für uns ableiten. Danke

**Reply by laske185** [reply, marked as answer]
.NET 6.0 ist die aktuelle LTS Version. Daher ist es nachvollziehbar, dass diese verwendet wird. Mit dem Release der nächsten LTS (8.0) sollte dann gewechselt werden, da wie bereits beschrieben, der Support immer auf drei Jahre limitiert ist. Durch die häufigen Releases seitens Microsoft sind die Migrationen auf neuere .NET Versionen relativ überschaubar – im Gegensatz zur Migration vom .NET Framework zu .NET über .NET Core.

Bei Neuentwicklungen könnten auch mit der Beta-Version von .NET 8 gearbeitet werden, wobei dies immer mit Vorsicht zu genießen ist.

Was .NET allgemein angeht, ist es ein modernes Framework u. a. mit C# als sehr mächtige Programmiersprache zum Entwickeln von Software in allen Bereichen, so auch für Web-Anwendungen. Es ist möglich, hierfür das Blazor-Framework zu verwenden, um Front- und Backend umzusetzen.

Blazor ist ein SPA-Framework, welches in zwei Varianten existiert: Blazor Server für Serverside Rendering und Blazor WebAssembly, bei der die gesamte Anwendung einschließlich der .NET-Runtime clientseitig im Browser ausgeführt wird und auch offline ausgeführt werden kann, wenn es einmal geladen war. Da Blazor Server per SignalR die Kommunikation mit dem Server realisiert wird, ist dieser schnell ein Bottleneck. Daher wird die Verwendung von Blazor WebAssembly eher empfohlen.

Alternativ kann .NET auch rein für das Backend z. B. zum Erstellen der REST-Services genutzt und das Frontend mit gängigen JavaScript-Frameworks wie React oder Vue.js umgesetzt werden.
